SConstruct

```diff
@@ -715,6 +715,21 @@ def get_soci_sources(style):
                        CPPPATH=cpp_path)
     return result
 
+def use_shp(toolchain):
+    '''
+    Return True if we want to use the --system-header-prefix command-line switch
+    '''
+    if toolchain != 'clang':
+        return False
+    if use_shp.cache is None:
+        try:
+            use_shp.cache = 'version 3.4' not in subprocess.check_output(
+                ['clang', '--version'], stderr=subprocess.STDOUT).strip()
+        except:
+            use_shp.cache = False
+    return use_shp.cache
+use_shp.cache = None
```

Comment:
Commenting on my own PR: A much nicer way to memoize the function is to use `@functools.lru_cache`. Unfortunately, we can't use that unless we upgrade to python 3. I also considered making my own decorator called `@static`, but decided the extra complexity wasn't worth it for just one function. If I use the same technique in other functions I'll revisit making a decorator.